common/meta/types/src/meta_storage_errors.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Display;
+
+use common_exception::ErrorCode;
+use serde::Deserialize;
+use serde::Serialize;
+use thiserror::Error;
+
+#[derive(Error, Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub enum MetaStorageError {
+    // type to represent serialize errors
+    #[error("{0}")]
+    SerializeError(String),
+
+    #[error("{0}")]
+    SerdeError(String),
+
+    #[error("{0}")]
+    SledError(String),
+
+    #[error("{0}")]
+    MetaStoreDamaged(String),
+
+    #[error("{0}")]
+    TransactionAbort(String),
+
+    #[error("{0}")]
+    TransactionError(String),
+
+    #[error("{0}")]
+    MetaSrvError(String),
```

Comment:
`MetaSrvError` is not used? I can not find any code using it:
![image](https://user-images.githubusercontent.com/44069/153000364-47c8f9c0-f632-4977-ab8f-6f95be8acb53.png)